pkg/auth/authenticator/password/ldappassword/ldap.go

```diff
@@ -88,18 +83,15 @@ func (a *Authenticator) getIdentity(username, password string) (authapi.UserIden
 		return nil, false, nil
 	}
 
-	// Make the connection
+	// Make the connection and bind to it if a bind DN and password were given
 	l, err := a.options.ClientConfig.Connect()
 	if err != nil {
 		return nil, false, err
 	}
 	defer l.Close()
 
-	// If specified, bind the username/password for search phase
-	if len(a.options.BindDN) > 0 {
-		if err := l.Bind(a.options.BindDN, a.options.BindPassword); err != nil {
-			return nil, false, err
-		}
+	if _, err := a.options.ClientConfig.Bind(l); err != nil {
```

Comment:
I understand where you're coming from, but this set of bind credentials is presumably necessary to get any useful data out of the LDAP server you're connecting to, that's why they are given - if a user were to see a set of LDAP client configs and *not* bind with these credentials, if they are present, they will get no data. This set of credentials seems quite distinct from the rest ...